src/Impostor.Server/Net/Manager/CompatibilityManager.cs

```diff
@@ -0,0 +1,100 @@
+using System.Collections.Generic;
+using Impostor.Api.Innersloth;
+using Impostor.Api.Net.Manager;
+using Microsoft.Extensions.Logging;
+
+namespace Impostor.Server.Net.Manager
+{
+    internal class CompatibilityManager : ICompatibilityManager
+    {
+        private static readonly CompatData[] DefaultSupportedVersions =
+        {
+            new(GameVersion.GetVersion(2022, 11, 1), GameVersion.GetVersion(2022, 11, 1), true), // 2022.12.8
+
+            new(GameVersion.GetVersion(2022, 11, 9), GameVersion.GetVersion(2022, 11, 9), true), // 2022.12.14
+
+            new(GameVersion.GetVersion(2022, 12, 2), GameVersion.GetVersion(2022, 12, 2), true), // 2023.2.28
+
+            new(GameVersion.GetVersion(2023, 1, 11), GameVersion.GetVersion(2023, 1, 11), true), // 2023.3.28s
+            new(GameVersion.GetVersion(2023, 3, 13), GameVersion.GetVersion(2023, 1, 11), true), // 2023.3.28a
+            new(GameVersion.GetVersion(2023, 4, 21), GameVersion.GetVersion(2023, 1, 11), true), // 2023.6.13
+
+            new(GameVersion.GetVersion(2023, 5, 20), GameVersion.GetVersion(2023, 5, 20), true), // 2023.7.11
+            new(GameVersion.GetVersion(2222, 0, 0), GameVersion.GetVersion(2023, 5, 20), false), // 2023.7.11 for host-only mods
+        };
```

Comment:
There's a lot of repetitiveness here. Maybe make the compat group an enum instead? Or store the game versions in static fields and reference them here

Reply:
If I'd make it an enum, removing a compatibility groups would be a breaking api change. Maybe it'd be better to make it an GameVersion array array where each element of the GameVersion array would be members of the same compatibility group?

Reply:
Sticking to this for now, yes it's a bit boilerplate but pretty reviewable